File: src/main/java/com/johnpier/labproject/models/BlogDto.java
package com.johnpier.labproject.models;

import lombok.*;

import java.io.Serializable;
import java.time.LocalDate;
import java.util.List;

@lombok.Data
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@EqualsAndHashCode
public class BlogDto implements Serializable {
    private String id;
    private String name;
    private String description;
    private EntitiesStatus status = null;
    private LocalDate createdAt;
    private List<PostPreviewDto> posts;
    private ShortUserDto createdBy;
    private List<CategoryDto> categories;
}

File: frontend/apps/blog-project/src/app/models/blog.dto.ts
import { CategoryDto } from './category.dto';
import { PostPreviewDto } from './post.dto';
import { ShortUserDto } from './user.dto';

export interface BlogPreviewDto {
  id: string;
  name: string;
  description: string;
  createdAt: string | Date;
  postsCount: number;
  createdBy: ShortUserDto;
  categories: any[];
}

export interface BlogDto {
  id: string;
  name: string;
  description: string;
  status: 'Active' | 'Archive';
  createdAt: string | Date;
  posts: PostPreviewDto[];
  createdBy: ShortUserDto;
  categories: CategoryDto[];
}

export interface BlogCreateDto {
  name: string;
  description: string;
  categories: string[];
}
